src/errors.ts

```diff
@@ -0,0 +1,82 @@
+import { register } from "./core";
+import { instrument } from "./instrumentation";
+
+let regexEscape = (str) => str.replace(/([.?*+^$&[\]\\(){}|<>-])/g, "\\$1");
+
+export function logError(errorToLog) {
+    // if errorToLog["error"] doesn't exist, occasionally we can get information directly from e
+    let error = errorToLog["error"] || errorToLog;
+    // this is a default value to indicate the object passed to us had no message member
+    let defaultMessage = "noMessage";
+    let source = errorToLog["filename"];
+    let lineno = errorToLog["lineno"];
+    let colno = errorToLog["colno"];
+    let message = error.message || defaultMessage;
+    let stack = error.stack;
+    let hrefRegex = new RegExp(regexEscape(location.href), "g");
+
+    if (stack) {
+        // ---------------------------------------------------------------------------------------------------------------------
+        // the stack has the potential to be very long with a lot of repeated filenames. The following is a small compression to
```

Comment:
I'm not very worried about the risk - the same compression has been running on Bing for a year or so. There's an argument to be made that the space saving is trivial compared to the amount of data the Clarity is already sending though - let me check what sorts of savings I'm actually getting and see what % of the payload they are.